src/models/model.cpp

```diff
@@ -884,79 +848,35 @@ void model::check_subgraph_parallelism()
 
 void model::setup_subgrid_layers_run_condition()
 {
-  // This function setups sub-grid run condition for a layer in forward and
-  // backward pass
   const auto& layers = this->get_layers();
   const El::Int num_layers = layers.size();
-  int myrank = El::mpi::Rank();
   for (El::Int node = 0; node < num_layers; ++node) {
-    auto layer_subgrid_ranks = layers[node]->get_subgrid_ranks();
-    if (layer_subgrid_ranks.find(myrank) != layer_subgrid_ranks.end() ||
-        layers[node]->get_name() == "layer1") {
-
-      layers[node]->set_run_layer_in_subgraph();
+    if(layers[node]->get_type() == "cross_grid_sum"){
```

Comment:
why are there special cases for layer types?

Reply:
Sub-graph parallelism is not a default case for these layers